core/modules/macros/contrastcolour.js

```diff
@@ -29,6 +29,22 @@ exports.params = [
 Run the macro
 */
 exports.run = function(target,fallbackTarget,colourA,colourB) {
```

Comment:
We can do that but it enormously uglifies the here proposed changes to the tag macro, by introducing (or keeping) a "quirk-macro" only to produce meaningful output, i.e. **tag-body-inner** alongside **tag-body** superfluously passing down a whole lot of arguments with the necessary variable setting just for sakes of proper evaluation of the kind that can easily be avoided... if only contrastcolor was a wee smarter.

I hope you have noticed that things have been heavily shifted away from the TagTemplate and into the tag macro so as to be able to reuse the same things in edit and view mode.

By the way, the way contrastcolor is set up right now only simplifies the case where you want it to be smart. You can still just pass colors to it as before and receive a contrast color.